rbm_softmax.py

```diff
@@ -0,0 +1,504 @@
+"""Restricted Boltzmann Machine with softmax visible units.
```

Comment:
Good call. [Done in a branch.](https://github.com/colinmorris/char-rbm/commit/7d643993f40ee07f153048df0c879413260ee019)

(That's the very latest version of the file, so it's a bit messier than the one in this P.R.)